From 62e70ab085ccc37ec9bb0d39bab74484df3d37c2 Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Wed, 26 Apr 2023 20:39:06 +0800
Subject: [PATCH] SWUpdate: upgrade fails by network. [1/1]

PD#SWPL-121506

Problem:
Upgrade fails by network.

Solution:
Kernel 5.4 uses /proc/inand nodes to determine
emmc or nand, which no longer works in kernel 5.15

Verify:
T7C PB

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: I015c21f2d80fbf7a2e1206e15ac6894ecb17f7b8
---
 bootloader_message/bootloader_message.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootloader_message/bootloader_message.c b/bootloader_message/bootloader_message.c
index 2976ac2..af44b40 100644
--- a/bootloader_message/bootloader_message.c
+++ b/bootloader_message/bootloader_message.c
@@ -558,7 +558,7 @@ static int set_bootloader_message_block(const struct bootloader_message *in) {
 
 int get_store_device() {
     int ret = 0;
-    ret = access("/proc/inand", F_OK);
+    ret = access("/sys/block/mmcblk0", F_OK);
     if (ret == 0 ) {
         printf("emmc device!\n");
         return 0;
-- 
2.25.1

